classes/class-swsales-reports.php

```diff
@@ -28,6 +29,36 @@ public static function add_reports_page() {
 		);
 	}
 
+	/**
+	 * Handles the Sales Export
+	 */
+	public static function sws_stats_csv() {
+		require_once(dirname(__FILE__) . "/../adminpages/report-csv.php");
+		exit;
+	}
+
+	/**
+	 * Gets a $csv_export_link for a sitewide sale.
+	 *
+	 * @param SWSales_Sitewide_Sale $sitewide_sale to get link for.
+	 * @return string $csv_export_link.
+	 * @since TBD.
+	 */
+	public static function build_CSV_report_link($sitewide_sale) {
+		//Bail if param is not correct.
+		if ( ! is_a( $sitewide_sale, 'Sitewide_Sales\classes\SWSales_Sitewide_Sale' ) ) {
+			return;
+		}
+		$csv_export_link = add_query_arg(
+			array(
+				'action' => 'sws_stats_csv',
+				'sitewide_sale' => $sitewide_sale->get_id(),
+			),
+			admin_url( 'admin-ajax.php' ) );
+
+		return esc_url( $csv_export_link );
```

Comment:
Probably better to not escape and return it raw and expect the code calling this to escape.

Okay to leave the esc_url here and ALSO esc_url when the URL is output. Running esc_url twice on the same URL should not break things.

Reply:
I removed the esc_url here and only escaping on output now.